esmvaltool/diag_scripts/land_carbon_cycle/diag_global_turnover.py

```diff
@@ -0,0 +1,822 @@
+                dpi=450)
+    plt.close()
+
+
+def main(diag_config):
+    '''
+    A function to calculate the modelled ecosystem carbon turnover time from
+    total carbon stock and gpp.
+
+    Arguments:
+        diag_config - nested dictionary of metadata
+    '''
+    my_files_dict = group_metadata(diag_config['input_data'].values(),
+                                   'dataset')
+
+    # get the data from the observation
+    global_tau_obs = _get_obs_data(diag_config)
+    base_name = ('{title}_{source_label}_'
+                 '{grid_label}'.format(
+                     title=global_tau_obs['grid']['tau_ctotal'].long_name,
+                     source_label=diag_config['obs_info']['source_label'],
+                     grid_label=diag_config['obs_info']['grid_label']))
+
+    global_tau_mod = {}
+    global_tau_mod['grid'] = {}
+    global_tau_mod['global'] = {}
+    plot_path_list = []
+    for key, value in my_files_dict.items():
+        global_tau_mod[key] = {}
+
+        # load the data
+        ctotal = _load_variable(value, 'ctotal')
+        gpp = _load_variable(value, 'gpp')
+        tau_ctotal = _calc_turnover(ctotal, gpp, diag_config, key)
+        global_tau_mod['grid'][key] = tau_ctotal
+
+        # apply the GPP threshold and set the data in dictionary
+        gpp_global = gpp.collapsed(['latitude', 'longitude'],
+                                   iris.analysis.SUM)
+        ctotal_global = ctotal.collapsed(['latitude', 'longitude'],
+                                         iris.analysis.SUM)
+        tau_global = ctotal_global / gpp_global
+        tau_global.convert_units('yr')
+
+        global_tau_mod['global'][key] = np.float(tau_global.core_data())
+
+        if diag_config['write_plots']:
+            base_name_mod = (
+                'global_{title}_{source_label}_'
+                '{grid_label}'.format(
+                    title=global_tau_obs['grid']['tau_ctotal'].long_name,
+                    source_label=key,
+                    grid_label=diag_config['obs_info']['grid_label']))
+            plot_path_mod = get_plot_filename(base_name_mod, diag_config)
+            plot_path_list.append(plot_path_mod)
+            _plot_single_map(plot_path_mod, tau_ctotal,
+                             global_tau_mod['global'][key], key, diag_config)
+
+    provenance_record = get_provenance_record(
+        "Comparison of global distributions of turnover time of carbon"
+        "with matrix comparison, maps, and multimodel bias. Reproduces"
+        "figure 3 in Carvalhais et al. (2014).", ['mean', 'perc'], ['global'],
+        _get_ancestor_files(diag_config, 'tau_ctotal'))
+
+    if diag_config['write_netcdf']:
+        model_cubes = [
+            c for c in global_tau_mod['grid'].values()
+            if isinstance(c, iris.cube.Cube)
+        ]
+        obs_cubes = [
+            c for c in global_tau_obs['grid'].values()
+            if isinstance(c, iris.cube.Cube)
+        ]
```

Comment:
could they be any other object than `iris.cube.Cube`?

Reply:
Sometimes, it can be numpy array.